### PR TITLE
Make the tolerance in step-1 relative.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: examples/step-1/step-1.cc [hunk 1]
```diff
@@ -210,14 +210,39 @@ void second_grid()
...
+              // can then also break the loop over all vertices and move on to
+              // the next cell.
+              //
+              // Because the distance from the center is computed as a floating
+              // point number, we have to expect that whatever we compute is
+              // only accurate to within
+              // [round-off](https://en.wikipedia.org/wiki/Round-off_error). As
+              // a consequence, we can never expect to compare the distance
+              // with the inner radius by equality: A statement such as
+              // `if (distance_from_center == inner_radius)` will always fail
```
Comment (tjhei) [inline examples/step-1/step-1.cc:224]:
```suggestion
              // `if (distance_from_center == inner_radius)` will fail
```

File: examples/step-1/step-1.cc [hunk 2]
```diff
@@ -210,14 +210,39 @@ void second_grid()
...
+              // unless we get exceptionally lucky. Rather, we need to do this
+              // comparison with a certain tolerance, and the usual way to do
+              // this is to write it as `if (std::abs(distance_from_center -
+              // inner_radius)
```
Comment (tjhei) [inline examples/step-1/step-1.cc:228]: remove newline?

**Review by masterleinad — APPROVED** [review]

**Review by tjhei — COMMENTED** [review]
Makes sense. Just some minor comments.

**Comment by bangerth** [comment]
@tjhei -- so done!

**Review by tjhei — APPROVED** [review]

**Pull request merged**